File: python/examples/format_from_bibtex.py
"""Format a BibTeX entry into multiple citation styles."""
from transtex import (
    format_apa,
    format_apa7,
    format_chicago,
    format_ieee,
    format_mla,
    format_vancouver,
    parse_bibtex_entry,
)


EXAMPLE = """@article{doe2020deep,
  author = {John Doe and Jane Smith},
  title = {Deep Learning for Everything},
  journal = {Journal of Omniscience},
  year = {2020},
  volume = {42},
  number = {7},
  pages = {1-10},
  doi = {10.1000/j.jo.2020.01.001},
  url = {https://example.com/article}
}"""


def main() -> None:
    reference = parse_bibtex_entry(EXAMPLE)
    print("APA 6th:")
    print(format_apa(reference))
    print()
    print("APA 7th:")
    print(format_apa7(reference))
    print()
    print("IEEE:")
    print(format_ieee(reference))
    print()
    print("MLA 9th:")
    print(format_mla(reference))
    print()
    print("Chicago (Author-Date):")
    print(format_chicago(reference))
    print()
    print("Vancouver:")
    print(format_vancouver(reference))


if __name__ == "__main__":
    main()

File: typescript/examples/formatFromBibtex.ts
// Run with: npx ts-node examples/formatFromBibtex.ts
import {
    formatApa,
    formatApa7,
    formatChicago,
    formatIeee,
    formatMla,
    formatVancouver,
    parseBibtexEntry,
} from "../src";

const EXAMPLE = `@article{doe2020deep,
  author = {John Doe and Jane Smith},
  title = {Deep Learning for Everything},
  journal = {Journal of Omniscience},
  year = {2020},
  volume = {42},
  number = {7},
  pages = {1-10},
  doi = {10.1000/j.jo.2020.01.001},
  url = {https://example.com/article}
}`;

function main(): void {
    const reference = parseBibtexEntry(EXAMPLE);
    console.log("APA 6th:");
    console.log(formatApa(reference));
    console.log("\nAPA 7th:");
    console.log(formatApa7(reference));
    console.log("\nIEEE:");
    console.log(formatIeee(reference));
    console.log("\nMLA 9th:");
    console.log(formatMla(reference));
    console.log("\nChicago (Author-Date):");
    console.log(formatChicago(reference));
    console.log("\nVancouver:");
    console.log(formatVancouver(reference));
}

main();
